### PR TITLE
Dept 5 search box UI

### PR DESCRIPTION
**Pull request by lsanchezcode**
https://www.figma.com/proto/bUrPkD0bOKPU1g2ikFrZOI/Shoutout---Rocket-(Kathleen-Copy)?node-id=107%3A1634&scaling=min-zoom&page-id=16%3A2&starting-point-node-id=125%3A3806&show-proto-sidebar=1

^ How the ui search box should look! 

Once you search, you will be brought to a 404 page, but if you inspect you can see in local storage the search.

### CODE REVIEW

**Review by bdurel — CHANGES_REQUESTED** [review]
Hey Lilli, I have a few notes mainly about styling. Nice work. I did have some notes that I left in the review, and have some over all ones I'll leave here. If you have any questions, please let me know. I think that the search box and button don't quite match the figma prototype for 1B. I'm attaching an image comparing the width of your searchbox (top) compared to the prototype (bottom). The width of the figma is 282.67px, and yours is 16.188rem(16px times 16.188rem = 259.008px). When sizing, take a look at the space between the magnifying glass and the left side of the search box. The prototype appears to have a wider space at 18.25px. I think that the clear icon might be a couple pixels off away from the right edge of the search box as well. The button also needs to be made larger to match the figma. 
image

**Comment by lsanchezcode** [comment]
@bdurel @Incarnate-Infinity I thinkkk this should be better! Did nesting where I could, it does not like nesting when it comes to things that change (like the cursor or colors or thing like that) so those will not be nested

**Comment by bdurel** [comment]
Hey Lilli, the search box is looking closer. There's currently no space between the bottom of the search component and the cards though. 
image
And I think it still doesn't match the figma. I lined them up and with yours on top here:

**Comment by bdurel** [comment]
Hey Lilli, nice work. HAve a few more comments though.
1. It's looking just about there to me, but the search box component appears to be off center.
image

2. The aria for the cancel botton is an off-center rectangle. Can we make it a square? 
image

3. I also have a very nitpicky one. The Cancel button seems to have more space on the right side of yours than it does on the figma. Maybe yours is matching the math on the figma though? The figma says 21.25px between the X (without any padding) and the input border. I can't quite tell what the math is on yours, but this is a screen comparison: 
image

**Review by Incarnate-Infinity — CHANGES_REQUESTED** [review]
The search box seems to be working great, found the results in session storage to be what I expected. It's looking pretty good, I've only got two major notes:

- Firstly I have is to second the search box/button being off center from the shoutout cards and the header/logo. I'm not sure what you've tried thus far, but it may be worth putting a little styling onto the form item wrapping the search box/button to see if we can size and center it properly, then just have the elements fall into place naturally from there.

- Second, I note that the `SearchBox.module.css` file is getting up into the repo. I assume this relates to how we're using some css properties for values that need to change, but I had thought we were also preventing the `.css` and `.css.map` files from getting up to the repo. Am I correct in thinking you need this file available to access some properties in it, or should it be getting kept out of the remote repository as well?

Other than those two things, this is looking awesome!

**Comment by lsanchezcode** [comment]
I tried to fix the box on the X when tabbed, I think that is just something that is part of the Icon? I can't find how to change the size of it without breaking the X

**Comment by nkhorvash** [comment]
Nice work, there's a lot of detail here and the styling matches Figma really well. 

I did find the button element seems to position a little to the right of the svg element for the 'X' button on inspection. I played with it a little and did not find a solution, so you may want to add a bug card for that specifically. I think everything else looks great.

**Comment by bdurel** [comment]
Hi Lilli, apart from what Nayson pointed out regarding responsiveness, it looks great, matches the figma's design perfectly and functions well. On the styling sheet I did see some SCSS nesting that could be done, specifically submitButtonActive and its two pseudoclasses could be written like this: 
`.submitButtonActive {
  --a: 45deg;
  transition: --a 300ms;
  background: linear-gradient(var(--a), $color-focus, #9873ff);
  color: #ffffff;
  cursor: pointer;
  &:hover {
    --a: -180deg;
  }
  &:active {
    background: #340f99;
  }
}`

I tried it out, and it didn't break anything on my end.

**Comment by nkhorvash** [comment]
One little thing, I almost forgot, can you delete the `.css` file? Is there any styling in the `.css` file not on the `.scss` file? We should have all the styling in the `.scss` format.

**Comment by lsanchezcode** [comment]
@nkhorvash I am not sure why it constantly tries to upload the `.css` file when I am only working in the `.scss`, I think maybe something to do with the complier? Maybe? But I will be sure to delete it before this last push!

**Comment by lsanchezcode** [comment]
Delete the css file, fixed the nesting, and added a bug fix for the weird button issue we were seeing!

File: components/search/search-box.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { MagnifyingGlassIcon, XMarkIcon } from '@heroicons/react/24/solid/';
+import { useRouter } from 'next/router';
+import { ChangeEvent, useRef, useState } from 'react';
+
+import fetch from '@/lib/fetch';
+
+import styles from './SearchBox.module.css';
```
Comment (nkhorvash) [inline components/search/search-box.tsx:7]: fix the import to .scss

**Comment by bdurel** [comment]
@lsanchezcode , I have a couple of new questions. The last bullet on the AC says "store search results in sessionStorage so they can be accessed when the search results page is implemented ([#DEPT-12](https://catalystit.atlassian.net/browse/DEPT-12)), and add a console.log for demo purposes (to be removed in [#DEPT-12](https://catalystit.atlassian.net/browse/DEPT-12))." When I search I'm getting a 404 and don't see anything in the console apart from the 404 not found error. Is that still part of the AC? I'm also wondering if there is supposed to be testing for this card? 
image

**Comment by lsanchezcode** [comment]
So general update comment! Added the console log and removed the routing to a new page as per the last AC to be fixed by the search results page!

**Comment by bdurel** [comment]
I believe that meets the AC for functionality on this card. Still curious about testing.

**Review by nkhorvash — APPROVED** [review]
The last minute fixes look good. @bdurel brings up a good point, about testing. Based on previous conversations, I would say it is implied, but also that the functional aspects of the card should be done first. This close to code freeze, my guess is PO would like to see the card pulled in and a tech debt card added for testing, but it doesn't hurt to check.

If my assumptions hold, everything else looks good to go to me.

**Comment by bdurel** [comment]
I just noticed another thing though. Can you rename the files? I'm seeing that a lot of the files in the project (not just in this component) aren't matching the team standards: search-box.tsx should be index.tsx, SearchBox.module.scss should be search.module.scss

**Review by bdurel — APPROVED** [review]

**Pull request merged**